misc/testerina/modules/testerina-core/src/main/ballerina/types.bal

```diff
@@ -15,15 +15,19 @@
 // under the License.
 
 # Possible return types of the data provider function.
-public type DataProviderReturnType error|map<AnyOrError[]>|AnyOrError[][];
+public type DataProviderReturnType error|map<AnyOrErrorOrReadOnlyType[]>|AnyOrErrorOrReadOnlyType[][];
 
-public type AnyOrError any|error;
+public type AnyOrErrorOrReadOnlyType any|error|ReadOnlyType;
+
+public type ReadOnlyType readonly;
```

Comment:
Let's have documentation comments for public types. Also we need to double check if these should be public.

Reply:
Imo, These should not be public.